### PR TITLE
direct implementations of some cliques related methods in `sage/graphs/graph.py`

### PR DESCRIPTION
**Pull request by dcoudert**
We propose direct implementations of the some methods related to maximal cliques to avoid calls to networkx.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

**Review by tscrim — COMMENTED** [review]
Should we leave an option to run the networkx implementation? In particular, how do the timings compare?

**Comment by dcoudert** [comment]
I did some tests and its surprising. This PR improves the situation for `cliques_get_max_clique_graph` but it is way slower for `cliques_get_clique_bipartite`. This is due to methods `add_edge` and `add_edges` of `BipartiteGraph` which are very slow (too many checks). So I should certainly revert changes in `cliques_get_clique_bipartite`.
```
sage: import networkx
sage: C = graphs.ChvatalGraph()
sage: F = graphs.FuzzyBallGraph([3,1],2)
sage: G = graphs.RandomBarabasiAlbert(100, 2)
sage: 
sage: %timeit C.cliques_get_max_clique_graph()
38.5 µs ± 109 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
sage: %timeit Graph(networkx.make_max_clique_graph(C.networkx_graph(), create_using=networkx.MultiGraph()), multiedges=False)
214 µs ± 1.82 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
sage: %timeit F.cliques_get_max_clique_graph()
16.2 µs ± 22 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
sage: %timeit Graph(networkx.make_max_clique_graph(F.networkx_graph(), create_using=networkx.MultiGraph()), multiedges=False)
43.7 µs ± 158 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
sage: %timeit G.cliques_get_max_clique_graph()
425 µs ± 1.48 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
sage: %timeit Graph(networkx.make_max_clique_graph(G.networkx_graph(), create_using=networkx.MultiGraph()), multiedges=False)
2.84 ms ± 7.4 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
sage: 
sage: %timeit C.cliques_get_clique_bipartite()
371 µs ± 581 ns per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
sage: %timeit BipartiteGraph(networkx.make_clique_bipartite(C.networkx_graph()))
144 µs ± 272 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
sage: %timeit F.cliques_get_clique_bipartite()
59 µs ± 149 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
sage: %timeit BipartiteGraph(networkx.make_clique_bipartite(F.networkx_graph()))
68.9 µs ± 108 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
sage: %timeit G.cliques_get_clique_bipartite()
13.4 ms ± 24.7 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
sage: %timeit BipartiteGraph(networkx.make_clique_bipartite(G.networkx_graph()))
993 µs ± 4.53 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
```

**Comment by dcoudert** [comment]
Clearly the slow down is due to method `add_edges` of `BipartiteGraph`. If I do as follows on the same graphs, I observe improvements. 
```
sage: def clique_bipartite(X):
....:     G = Graph([X, []], format='vertices_and_edge')
....:     for i, clique in enumerate(IndependentSets(X, maximal=True, complement=True)):
....:         idx = - i - 1
....:         G.add_vertex(idx)
....:         G.add_edges((u, idx) for u in clique)
....:     return BipartiteGraph(G)
sage:
sage: %timeit clique_bipartite(C)
98.4 µs ± 550 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
sage: %timeit clique_bipartite(F)
42.6 µs ± 287 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
sage: %timeit clique_bipartite(G)
727 µs ± 4.21 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
```

File: src/sage/graphs/graph.py [hunk 1]
```diff
@@ -6996,18 +7024,32 @@ def cliques_get_clique_bipartite(self, **kwds):
...
-        from .bipartite_graph import BipartiteGraph
-        import networkx
-        return BipartiteGraph(networkx.make_clique_bipartite(self.networkx_graph(), **kwds))
+        G = Graph([self, []], format='vertices_and_edges')
+        for i, clique in enumerate(IndependentSets(self, maximal=True, complement=True)):
+            idx = - i - 1
+            G.add_vertex(idx)
+            G.add_edges((u, idx) for u in clique)
+        from sage.graphs.bipartite_graph import BipartiteGraph
+        return BipartiteGraph(G)
```
Comment (tscrim) [inline src/sage/graphs/graph.py]:
```suggestion
        return BipartiteGraph(G, check=False)
```

**Review by tscrim — APPROVED** [review]
Thanks. LGTM.

**Comment by dcoudert** [comment]
Thanks for the review.

**Pull request merged**